### PR TITLE
Allow datapoints to be Objects

### PR DESCRIPTION
**Pull request by vladcostea**
Each datapoint in the data Array of a dataset can now be an Object. All
it needs is a `y` property with the actual value for that point.

The `isObject` function is lifted from the underscore lib.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,17 @@
 (function(Chart) {
+  var isObject = function(obj) {
+    var type = typeof obj;
+    return type === 'object' && !!obj;
+  }
+
+  var dataValue = function(dataPoint) {
+    if (isObject(dataPoint)) {
+      return dataPoint.y;
```
Comment (y-takey) [inline src/index.js]: I think it needs to support `horizontalBar`.
When chart type is `horizontalBar`, datapoints object has `x` property.

```js
new Chart(..., {
  type: "horizontalBar",
  data: {
    ...,
    datasets: [
      { data: [{ x: 1 }, { x: 2 }, { x: 0 }] },
    ]
  },
  ...
}
```

**Comment by y-takey** [comment]
Hi @vladcostea , Thanks for the PR! sounds great :)
I think there is only one issue. Please look at the above my comment.

File: src/index.js (hunk 1 above)
Comment (vladcostea) [reply src/index.js]: You're right. I'll try to fix it soon, thanks.

**Comment by y-takey** [comment]
@vladcostea LGTM :) Thank you for the quick response!
I've merged it and publish newer npm module(v0.6.0). thanks.

**Comment by vladcostea** [comment]
@y-takey with pleasure :)

I'm sure this could've been done in a more generic way (have a factory that builds a datapoint value fetcher for each type of chart), however that felt like overkill.

I also hope this is the Chart.js "way" of getting the data from a datapoint and I'm not introducing a performance hit with all the extra checks (some is inevitable, but acceptable imho)

**Pull request merged**